pgjdbc/src/test/java/org/postgresql/test/jdbc2/DriverTest.java

```diff
@@ -62,6 +62,7 @@ public void testAcceptsURL() throws Exception {
     verifyUrl(drv, "jdbc:postgresql://127.0.0.1/anydbname", "127.0.0.1", "5432", "anydbname");
     verifyUrl(drv, "jdbc:postgresql://127.0.0.1:5433/hidden", "127.0.0.1", "5433", "hidden");
     verifyUrl(drv, "jdbc:postgresql://[::1]:5740/db", "[::1]", "5740", "db");
+    verifyUrl(drv, "jdbc:postgresql:?service=testsrv", "srvhost", "5434", "srvdb");
```

Comment:
Thanks for working on this, @bersace !

Looking at the docs for [libpq connection URIs](https://www.postgresql.org/docs/11/static/libpq-connect.html#id-1.7.3.8.3.6):

```
postgresql://
postgresql://localhost
postgresql://localhost:5433
postgresql://localhost/mydb
postgresql://user@localhost
postgresql://user:secret@localhost
postgresql://other@localhost/otherdb?connect_timeout=10&application_name=myapp
postgresql://host1:123,host2:456/somedb?target_session_attrs=any&application_name=myapp
```

The syntax for specifying only a service would be one of
 - `postgresql://?service=SERVICE`
 - `postgresql:///?service=SERVICE`

The syntax `postgresql:?service=SERVICE` and `postgresql:?service=SERVICE` both fail (tested with `psql` version 10.4). Using two or three slashes works.

There's a note in the docs:

> For improved compatibility with JDBC connection URIs, instances of parameter `ssl=true` are translated into `sslmode=require`.

I think we should do the same and try to be compatible with libpq syntax here.

Reply:
Hi @grzm, thanks for the review !

That's fair. Updated ! :-)